test/controllers/DeclarationDetailsControllerSpec.scala

```diff
@@ -0,0 +1,143 @@
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import views.html.{declaration_details, declaration_information}
+
+class DeclarationDetailsControllerSpec extends ControllerWithoutFormSpec with BeforeAndAfterEach {
+
+  private val actionId = "actionId"
+  private val notification = Notification(actionId, "mrn", ZonedDateTime.now, SubmissionStatus.UNKNOWN, Seq.empty)
+
+  private val submission = Submission(
+    uuid = UUID.randomUUID().toString,
+    eori = "eori",
+    lrn = "lrn",
+    mrn = None,
+    ducr = None,
+    actions = Seq(Action(id = actionId, requestType = SubmissionRequest, requestTimestamp = ZonedDateTime.now))
+  )
+
+  private val queryNotificationMessageConfig = mock[QueryNotificationMessageConfig]
+  private val declarationInformationPage = mock[declaration_information]
+  private val declarationDetailsPage = mock[declaration_details]
+
+  val controller = new DeclarationDetailsController(
+    mockAuthAction,
+    mockVerifiedEmailAction,
+    mockCustomsDeclareExportsConnector,
+    stubMessagesControllerComponents(),
+    queryNotificationMessageConfig,
+    declarationInformationPage,
+    declarationDetailsPage
+  )(ec)
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+
+    authorizedUser()
+    when(declarationInformationPage.apply(any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+    when(declarationDetailsPage.apply(any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit =
+    reset(queryNotificationMessageConfig, declarationInformationPage, declarationDetailsPage, mockCustomsDeclareExportsConnector)
+
+  "Display Submissions" should {
+
+    "return 200 (OK)" when {
+
+      "display declaration with notification method is invoked" when {
+
+        "QueryNotificationMessage feature flag is disabled" in {
+
+          responsesToReturn(isQueryNotificationMessageEnabled = false)
+
+          val result = controller.displayPage(actionId)(getRequest())
+          status(result) mustBe OK
+
+          verifyNoInteractions(declarationDetailsPage)
+
+          verifyPage { (submissionCaptor: ArgumentCaptor[Submission], notificationsCaptor: ArgumentCaptor[Seq[Notification]]) =>
+            verify(declarationInformationPage).apply(submissionCaptor.capture(), notificationsCaptor.capture())(any(), any())
+          }
+        }
+
+        "QueryNotificationMessage feature flag is enabled" in {
+
+          responsesToReturn(isQueryNotificationMessageEnabled = true)
+
+          val result = controller.displayPage(actionId)(getRequest())
+          status(result) mustBe OK
+
+          verifyNoInteractions(declarationInformationPage)
+
+          verifyPage { (submissionCaptor: ArgumentCaptor[Submission], notificationsCaptor: ArgumentCaptor[Seq[Notification]]) =>
+            verify(declarationDetailsPage).apply(submissionCaptor.capture(), notificationsCaptor.capture())(any(), any())
+          }
```

Comment:
Nit: I am not convinced with this new approach. We used to handle this situation by having a method returning proper `Captor` instance. And then it was invoked for every verify I am just not comfortable with introducing more and more new approaches to handle the same situation.
Maybe instead of passing this function parameter, you could simply pass the template/page? You'd avoid some of the boiler-plate code.

Reply:
I tried just passing the template/page, but what's the common type to be used for the page parameter?
If you have an alternative implementation I'll be more than happy to use that. I really do not care about how we implement this verification. If not I can even just use the same verifying code for each of two pages under test.

Reply:
Refactored